statistics/handler/dump.go

```diff
@@ -0,0 +1,364 @@
+			if err != nil {
+				return errors.Trace(err)
+			}
+		}
+	}
+	return errors.Trace(h.Update(is))
+}
+
+func (h *Handle) loadStatsFromJSON(tableInfo *model.TableInfo, physicalID int64, jsonTbl *JSONTable) error {
+	tbl, err := TableStatsFromJSON(tableInfo, physicalID, jsonTbl)
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	for _, col := range tbl.Columns {
+		err = h.SaveStatsToStorage(tbl.PhysicalID, tbl.Count, 0, &col.Histogram, col.CMSketch, 1)
+		if err != nil {
+			return errors.Trace(err)
+		}
+	}
+	for _, idx := range tbl.Indices {
+		err = h.SaveStatsToStorage(tbl.PhysicalID, tbl.Count, 1, &idx.Histogram, idx.CMSketch, 1)
+		if err != nil {
+			return errors.Trace(err)
+		}
+	}
+	err = h.SaveMetaToStorage(tbl.PhysicalID, tbl.Count, tbl.ModifyCount)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	return nil
+}
+
+// TableStatsFromJSON loads statistic from JSONTable and return the Table of statistic.
+func TableStatsFromJSON(tableInfo *model.TableInfo, physicalID int64, jsonTbl *JSONTable) (*statistics.Table, error) {
+	newHistColl := statistics.HistColl{
+		PhysicalID:     physicalID,
+		HavePhysicalID: true,
+		Count:          jsonTbl.Count,
+		ModifyCount:    jsonTbl.ModifyCount,
+		Columns:        make(map[int64]*statistics.Column, len(jsonTbl.Columns)),
+		Indices:        make(map[int64]*statistics.Index, len(jsonTbl.Indices)),
+	}
+	tbl := &statistics.Table{
+		HistColl: newHistColl,
+	}
+	for id, jsonIdx := range jsonTbl.Indices {
+		for _, idxInfo := range tableInfo.Indices {
+			if idxInfo.Name.L != id {
+				continue
+			}
+			hist := statistics.HistogramFromProto(jsonIdx.Histogram)
+			hist.ID, hist.NullCount, hist.LastUpdateVersion = idxInfo.ID, jsonIdx.NullCount, jsonIdx.LastUpdateVersion
+			idx := &statistics.Index{
+				Histogram: *hist,
+				CMSketch:  statistics.CMSketchFromProto(jsonIdx.CMSketch),
+				Info:      idxInfo,
+			}
+			tbl.Indices[idx.ID] = idx
+		}
+	}
+
+	for id, jsonCol := range jsonTbl.Columns {
+		for _, colInfo := range tableInfo.Columns {
+			if colInfo.Name.L != id {
+				continue
+			}
+			hist := statistics.HistogramFromProto(jsonCol.Histogram)
+			count := int64(hist.TotalRowCount())
+			sc := &stmtctx.StatementContext{TimeZone: time.UTC}
+			hist, err := hist.ConvertTo(sc, &colInfo.FieldType)
+			if err != nil {
+				return nil, errors.Trace(err)
+			}
+			hist.ID, hist.NullCount, hist.LastUpdateVersion, hist.TotColSize = colInfo.ID, jsonCol.NullCount, jsonCol.LastUpdateVersion, jsonCol.TotColSize
+			col := &statistics.Column{
+				PhysicalID: physicalID,
+				Histogram:  *hist,
+				CMSketch:   statistics.CMSketchFromProto(jsonCol.CMSketch),
+				Info:       colInfo,
+				Count:      count,
+				IsHandle:   tableInfo.PKIsHandle && mysql.HasPriKeyFlag(colInfo.Flag),
+			}
+			tbl.Columns[col.ID] = col
+		}
+	}
+	return tbl, nil
+}
+
+func (h *Handle) dumpRangeFeedback(sc *stmtctx.StatementContext, ran *ranger.Range, rangeCount float64, q *statistics.QueryFeedback) error {
```

Comment:
This file should be for `dump stats`. Maybe it is better to move it to another file?